packages/lib/src/components/AmazonPay/AmazonPay.tsx

```diff
@@ -0,0 +1,59 @@
+import { h } from 'preact';
+import UIElement from '../UIElement';
+import CoreProvider from '../../core/Context/CoreProvider';
+import collectBrowserInfo from '../../utils/browserInfo';
+import AmazonPayComponent from './components/AmazonPayComponent';
+import { AmazonPayElementData, AmazonPayElementProps } from './types';
+import defaultProps from './defaultProps';
+import './AmazonPay.scss';
+
+export class AmazonPayElement extends UIElement<AmazonPayElementProps> {
+    public static type = 'amazonpay';
+    protected static defaultProps = defaultProps;
+
+    formatProps(props) {
+        return {
+            ...props,
+            environment: props.environment.toUpperCase(),
+            locale: props.locale.replace('-', '_'),
```

Comment:
Have we considered using `this.props.i18n.locale` instead? (The resolved locale, for example: `es-AR` => `es-ES`) Or are do we really want the locale as specified by the merchant?

Reply:
Both the Amazon Pay SDK and the PayPal SDK accept a different subset of locales (see `packages/lib/src/components/AmazonPay/config.ts`).
We could use `this.props.i18n.locale` and transform `en_US` to `en_GB` for the `EU` region but I don't know if it would unclear for merchants. What do you think?